bot/skills/length.py

```diff
@@ -0,0 +1,144 @@
+import logging
+from typing import Optional, List, TypedDict
+
+from telegram import Update, User, Message, ParseMode
+from telegram.ext import Updater, CommandHandler, CallbackContext
+from telegram.ext.filters import Filters
+from telegram.utils.helpers import mention_markdown
+
+from mode import cleanup_queue_update
+from config import get_group_chat_id
+
+from skills.roll import _get_username
+
+from db.mongo import get_db
+import pymongo
+from pymongo.collection import Collection
+from pymongo.results import UpdateResult
+
+logger = logging.getLogger(__name__)
+
+PENINSULA_LENGTH_FIELD = "peninsula_length"
+PENINSULA_MAXIMAL_LENGTH = 20
+
+
+class PeninsulaDataType(TypedDict):
+    _id: str
+    peninsula_length: int
+    meta: User
+
+
+def add_length(upd: Updater, handlers_group: int):
+    logger.info("registering length handlers")
+    dp = upd.dispatcher
+    dp.add_handler(
+        CommandHandler(
+            "length",
+            _length,
+            filters=Filters.chat(username=get_group_chat_id().strip("@")),
+            run_async=True,
+        ),
+        handlers_group,
+    )
+
+    dp.add_handler(
+        CommandHandler(
+            "longest",
+            _longest,
+            filters=Filters.chat(username=get_group_chat_id().strip("@")),
+            run_async=True,
+        ),
+        handlers_group,
+    )
+
+
+class DB:
+    def __init__(self, db_name: str):
+        self._coll: Collection = get_db(db_name).peninsulas
+
+    def get_longest_peninsula(self) -> int:
+        result = PENINSULA_MAXIMAL_LENGTH
+        x: PeninsulaDataType = self._coll.find_one(
+            sort=[(PENINSULA_LENGTH_FIELD, pymongo.DESCENDING)]
+        )
+        if x is not None:
+            result = max(result, int(x[PENINSULA_LENGTH_FIELD]))
+        return result
+
+    def get_best_n(self, n: int = 10) -> List[PeninsulaDataType]:
+        return list(
+            self._coll.find({}).limit(n).sort(PENINSULA_LENGTH_FIELD, pymongo.ASCENDING)
```

Comment:
Shouldn't sort come after limit?

Reply:
Yes, by common sense it is and that is a mistype. But actually in pymongo calling limit does nothing but sets limit in query (https://github.com/mongodb/mongo-python-driver/blob/c760f900f2e4109a247c2ffc8ad3549362007772/pymongo/cursor.py#L491), so query should work the same way no matter how you order a calls here.